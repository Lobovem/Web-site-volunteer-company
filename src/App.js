import s from './App.module.scss';

function App() {
  return <div className={s.App}>TEST</div>;
}

export default App;
