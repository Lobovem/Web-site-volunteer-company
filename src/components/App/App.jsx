import { useEffect } from 'react';
import { PageHome } from '../PageHome/PageHome';
import { Header } from '../Header/Header';
import { Footer } from '../Footer/Footer';
import { PageAboutUs } from '../PageAboutUs/PageAboutUs';
import { PageNews } from '../PageNews/PageNews';
import { PageError } from '../PageError/PageError';
import { PageOneNews } from '../PageSimpleNews/PageOneNews';
import { PageGetHelp } from '../PageGetHelp/PageGetHelp';
import { useDispatch, useSelector } from 'react-redux';
import { ScrollToTop } from '../ScrollTop/ScrollTop';
import { PageUserCondition } from '../PageUserConditions/PageUserCondition';
import { PageContacts } from '../PageContacts/PageContacts';
import { PageDonate } from '../PageDonate/PageDonate';
import { Loader } from '../Loader/Loader';
import { fetchMenu, fetchNews } from '../../api/api';
import s from './app.module.scss';
import { Route, Routes } from 'react-router-dom';

export const App = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMenu());
    dispatch(fetchNews());
  }, [dispatch]);

  const isLoading = useSelector((state) => state.content.isLoading);
  const error = useSelector((state) => state.content.error);

  if (isLoading) {
    return (
      <div className={s.loader}>
        <Loader />
      </div>
    );
  }

  if (error) {
    return <h1 className={s.error}>{error}</h1>;
  }

  return (
    <div>
      <ScrollToTop />
      <Header />
      <Routes>
        <Route path="/" element={<PageHome />} />
        <Route path="/about-us" element={<PageAboutUs />} />
        <Route path="/news" element={<PageNews />}></Route>
        <Route path="/news/:id" element={<PageOneNews />} />
        <Route path="/get-help" element={<PageGetHelp />} />
        <Route path="/donate" element={<PageDonate />} />
        <Route path="/contacts" element={<PageContacts />} />
        <Route path="/user-condition" element={<PageUserCondition />} />
        <Route path="*" element={<PageError />} />
      </Routes>
      <Footer />
    </div>
  );
};
