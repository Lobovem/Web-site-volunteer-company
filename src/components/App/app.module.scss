.body {
  &.active {
    overflow: hidden;

  }
}

.loader {
  min-height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.error {
  padding: 50px;
  font-size: 30px;
  text-align: center;
}