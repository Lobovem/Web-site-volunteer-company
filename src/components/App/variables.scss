//fonts-family
$font-family: 'Rubik', sans-serif;

//fonts-weight
$font-weight-400: 400;
$font-weight-500: 500;
$font-weight-600: 600;

//font-size
$font-size-14: 14px;
$font-size-18: 18px;
$font-size-24: 24px;
$font-size-38: 38px;

//color
$color-white: #fff;
$color-main: #025FA7;
$color-grey: #F3F3F3;
$color-grey-light: #025fa714;
$color-yellow: #FFD500;
$color-red: #C82E37;
$color-green: #36B98F;

//background
$gradient-main: linear-gradient(99.7deg, #025FA7 5.07%, #1682D4 32.97%, #00B0D0 60.87%, #09D3F6 88.76%, #42E3FF 116.66%);
$gradient-hover: linear-gradient(98.35deg, #025FA7 5.18%, #1682D4 46.68%, #00B0D0 88.18%, #09D3F6 129.69%, #42E3FF 171.19%);