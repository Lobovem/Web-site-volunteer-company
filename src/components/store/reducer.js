import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  burgerMenuState: false,
};

export const appReducer = createSlice({
  name: 'app',
  initialState: initialState,
  reducers: {
    isOpenBurgerMenu: (state, action) => {
      state.burgerMenuState = !state.burgerMenuState;
    },
  },
});

export const burgerMenuSelector = (state) => state.app.burgerMenuState;

export const { isOpenBurgerMenu } = appReducer.actions;
export default appReducer.reducer;
