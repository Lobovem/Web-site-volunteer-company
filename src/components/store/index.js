// import configureStore from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { appReducer } from './reducer';

export const store = configureStore({
  reducer: {
    todoList: appReducer.reducer,
  },
});
