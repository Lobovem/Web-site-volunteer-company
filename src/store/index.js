// import configureStore from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { todoReducer } from './reducer';

export const store = configureStore({
  reducer: {
    todoList: todoReducer.reducer,
  },
});
