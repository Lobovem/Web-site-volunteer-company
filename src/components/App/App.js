import { useSelector } from 'react-redux';
import { burgerMenuSelector } from '../store/reducer';
import { Header } from '../Header/Header';
import { Banner } from '../Banner/Banner';
import { useEffect } from 'react';
import { AboutUs } from '../AboutUs/AboutUs';
import { News } from '../News/News';
import { OurResult } from '../OurResult/OurResult';
import { WaySupport } from '../WaySupport/WaySupport';
import { Donate } from '../Donate/Donate';
import { Teams } from '../Teams/Teams';
import { Contacts } from '../Contacts/Contacts';
import { Form } from '../Form/Form';

function App() {
  const burgerMenu = useSelector(burgerMenuSelector);

  useEffect(() => {
    if (burgerMenu) {
      document.body.classList.add('active');
    } else {
      document.body.classList.remove('active');
    }
  }, [burgerMenu]);

  return (
    <div>
      <Header></Header>
      <Banner></Banner>
      <AboutUs></AboutUs>
      <News></News>
      <OurResult></OurResult>
      <WaySupport></WaySupport>
      <Donate></Donate>
      <Teams></Teams>
      <Contacts></Contacts>
      <Form></Form>
    </div>
  );
}

export default App;
