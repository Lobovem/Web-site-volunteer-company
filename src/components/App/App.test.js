import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider, useDispatch, useSelector } from 'react-redux';
import { store } from '../../redux/config/store';
import { App } from './App';

jest.mock('../../api/api', () => ({
  fetchMenu: jest.fn(),
  fetchNews: jest.fn(),
}));

jest.mock('react-redux', () => {
  const originalModule = jest.requireActual('react-redux');
  return {
    ...originalModule,
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
  };
});

describe('App', () => {
  test('renders App component', () => {
    const dispatch = jest.fn();
    useDispatch.mockReturnValue(dispatch);
    // useSelector.mockReturnValue({ isLoading: false, error: null });

    const { container } = render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
