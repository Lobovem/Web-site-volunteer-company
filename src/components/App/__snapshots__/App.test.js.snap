// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders app component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header layout"
      >
        <div
          class="header__wrap"
        >
          <div
            class="logoMain"
          >
            <img
              alt="ukraine-flag.png"
              class="logoMain__img"
              src="logo-main.svg"
            />
          </div>
          <div
            class="header__menuWrap"
          >
            <div
              class="burger"
            >
              <span
                class="burger__line"
              />
              <span
                class="burger__line"
              />
              <span
                class="burger__line"
              />
            </div>
            <nav
              class="menu"
            >
              <ul
                class="menu__list"
              >
                <li
                  class="menu__item"
                >
                  <a
                    class="menu__link"
                    href="/about"
                  >
                    Про нас
                  </a>
                </li>
                <li
                  class="menu__item"
                >
                  <a
                    class="menu__link"
                    href="/news"
                  >
                    Новини
                  </a>
                </li>
                <li
                  class="menu__item"
                >
                  <a
                    class="menu__link"
                    href="/get-help"
                  >
                    Отримати допомогу
                  </a>
                </li>
                <li
                  class="menu__item"
                >
                  <a
                    class="menu__link"
                    href="/help"
                  >
                    Допомогти
                  </a>
                </li>
                <li
                  class="menu__item"
                >
                  <a
                    class="menu__link"
                    href="/contacts"
                  >
                    Контакти
                  </a>
                </li>
              </ul>
            </nav>
            <div
              class="langWrap"
            >
              <a
                class="langWrap__link langWrap__link_active"
                href="/"
              >
                UA
              </a>
              <a
                class="langWrap__link"
                href="/"
              >
                EN
              </a>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header layout"
    >
      <div
        class="header__wrap"
      >
        <div
          class="logoMain"
        >
          <img
            alt="ukraine-flag.png"
            class="logoMain__img"
            src="logo-main.svg"
          />
        </div>
        <div
          class="header__menuWrap"
        >
          <div
            class="burger"
          >
            <span
              class="burger__line"
            />
            <span
              class="burger__line"
            />
            <span
              class="burger__line"
            />
          </div>
          <nav
            class="menu"
          >
            <ul
              class="menu__list"
            >
              <li
                class="menu__item"
              >
                <a
                  class="menu__link"
                  href="/about"
                >
                  Про нас
                </a>
              </li>
              <li
                class="menu__item"
              >
                <a
                  class="menu__link"
                  href="/news"
                >
                  Новини
                </a>
              </li>
              <li
                class="menu__item"
              >
                <a
                  class="menu__link"
                  href="/get-help"
                >
                  Отримати допомогу
                </a>
              </li>
              <li
                class="menu__item"
              >
                <a
                  class="menu__link"
                  href="/help"
                >
                  Допомогти
                </a>
              </li>
              <li
                class="menu__item"
              >
                <a
                  class="menu__link"
                  href="/contacts"
                >
                  Контакти
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="langWrap"
          >
            <a
              class="langWrap__link langWrap__link_active"
              href="/"
            >
              UA
            </a>
            <a
              class="langWrap__link"
              href="/"
            >
              EN
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
